File: app/src/main/java/de/xikolo/controllers/announcement/AnnouncementActivity.java
package de.xikolo.controllers.announcement;

import android.os.Bundle;
import android.support.design.widget.AppBarLayout;
import android.support.design.widget.CollapsingToolbarLayout;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.view.View;
import android.widget.ImageView;

import com.yatatsu.autobundle.AutoBundleField;

import butterknife.BindView;
import de.xikolo.R;
import de.xikolo.config.GlideApp;
import de.xikolo.controllers.base.BaseActivity;
import de.xikolo.controllers.helper.CollapsingToolbarHelper;
import de.xikolo.models.Announcement;
import de.xikolo.models.Course;

import static de.xikolo.R.id.appbar;
import static de.xikolo.R.id.collapsing_toolbar;

public class AnnouncementActivity extends BaseActivity {

    public static final String TAG = AnnouncementActivity.class.getSimpleName();

    @AutoBundleField String announcementId;
    @AutoBundleField boolean global;

    @BindView(R.id.toolbar_image) ImageView imageView;
    @BindView(appbar) AppBarLayout appBarLayout;
    @BindView(collapsing_toolbar) CollapsingToolbarLayout collapsingToolbar;
    @BindView(R.id.scrim_top) View scrimTop;
    @BindView(R.id.scrim_bottom) View scrimBottom;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_blank_collapsing);
        setupActionBar(true);
        enableOfflineModeToolbar(false);

        Announcement announcement = Announcement.get(announcementId);
        setTitle(announcement.title);

        if (announcement.imageUrl != null) {
            GlideApp.with(this).load(announcement.imageUrl).into(imageView);
        } else if (announcement.courseId != null) {
            Course course = Course.get(announcement.courseId);
            if (course.imageUrl != null) {
                GlideApp.with(this).load(course.imageUrl).into(imageView);
            } else {
                CollapsingToolbarHelper.lockCollapsingToolbar(announcement.title, appBarLayout, collapsingToolbar, toolbar, scrimTop, scrimBottom);
            }
        } else {
            CollapsingToolbarHelper.lockCollapsingToolbar(announcement.title, appBarLayout, collapsingToolbar, toolbar, scrimTop, scrimBottom);
        }

        String tag = "content";

        FragmentManager fragmentManager = getSupportFragmentManager();
        if (fragmentManager.findFragmentByTag(tag) == null) {
            FragmentTransaction transaction = fragmentManager.beginTransaction();
            transaction.replace(R.id.content, AnnouncementFragmentAutoBundle.builder(announcementId, global).build(), tag);
            transaction.commit();
        }
    }

}

File: app/src/main/java/de/xikolo/controllers/announcement/AnnouncementActivity.kt
package de.xikolo.controllers.announcement

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import com.yatatsu.autobundle.AutoBundleField
import de.xikolo.R
import de.xikolo.config.GlideApp
import de.xikolo.controllers.base.CollapsingToolbarActivity
import de.xikolo.controllers.dialogs.CreateTicketDialog
import de.xikolo.controllers.dialogs.CreateTicketDialogAutoBundle
import de.xikolo.models.dao.AnnouncementDao

class AnnouncementActivity : CollapsingToolbarActivity() {

    companion object {
        val TAG: String = AnnouncementActivity::class.java.simpleName
    }

    @AutoBundleField
    lateinit var announcementId: String

    @AutoBundleField
    var global: Boolean = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        AnnouncementDao.Unmanaged.find(announcementId)?.let { announcement ->
            title = announcement.title

            if (announcement.imageUrl != null) {
                GlideApp.with(this).load(announcement.imageUrl).into(imageView)
            } else if (announcement.courseId != null) {
                val course = announcement.course
                if (course?.imageUrl != null) {
                    GlideApp.with(this).load(course.imageUrl).into(imageView)
                } else {
                    lockCollapsingToolbar(announcement.title)
                }
            } else {
                lockCollapsingToolbar(announcement.title)
            }

            val tag = "content"

            val fragmentManager = supportFragmentManager
            if (fragmentManager.findFragmentByTag(tag) == null) {
                val transaction = fragmentManager.beginTransaction()
                transaction.replace(
                    R.id.content,
                    AnnouncementFragmentAutoBundle.builder(announcementId, global).build(),
                    tag
                )
                transaction.commit()
            }
        }

    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.helpdesk, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.action_helpdesk -> {
                val dialog = CreateTicketDialogAutoBundle.builder().build()
                dialog.show(supportFragmentManager, CreateTicketDialog.TAG)
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }
}
